test/integration/au/org/emii/aatams/detection/DetectionFactoryServiceIntegrationTests.groovy

```diff
@@ -12,15 +12,15 @@ class DetectionFactoryServiceIntegrationTests extends GroovyTestCase {
     // Test for #1751
     void testRescanForDeployment()
     {
-        def initValidCount = ValidDetection.count()
-        def initInvalidCount = InvalidDetection.count()
-
         def sql = new Sql(dataSource)
         ValidDetection.metaClass.static.count = {
 
             return sql.firstRow('select count(*) from valid_detection').count
         }
 
+        def initValidCount = ValidDetection.count()
+        def initInvalidCount = InvalidDetection.count()
+
```

Comment:
I had to move these calls to count() to after we add our own version of count() to the ValidDetection metaclass.